base/util.go

```diff
@@ -1259,6 +1259,16 @@ func injectJSONPropertyFromBytes(b []byte, bIsEmpty bool, kvPairs []KVPairBytes)
 	return newJSON
 }
 
+func UnknownFieldErrCheck(err error) error {
+	if err != nil && strings.Contains(err.Error(), "unknown field") {
+		// Special handling for unknown field errors
+		// json.Decode continues to decode the full data into the struct
+		// so it's safe to use even after this error
```

Comment:
This comment was relevant in the piece of code that this was pulled from, but not really relevant here.